rest-service/manager_rest/resource_manager.py

```diff
@@ -2317,13 +2370,37 @@ def _create_deployment_initial_dependencies(self,
                             fail_silently=True) if target_deployment else None
 
             now = utils.get_formatted_timestamp()
+            if external_client:
```

Comment:
How about separating this part (that puts a new dependency) in a new function, based on `external_client`? 
i.e. one function would be `put_new_dependency_with_external_client` and the other would be `put_new_dependency`? Of course, the functions' names are only a suggestion.

Reply:
much of the code is common to both...

Reply:
Than it should be in a function that they will both use? "If there is a flag in the function it should be two functions" - Lukasz :)